### ISSUE TITLE
no-extra-parens false positive with await and exponentiation operator

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12739]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2017,
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogXCJlcnJvclwiKi9cblxuYXN5bmMgZnVuY3Rpb24gZm9vICgpIHtcbiAgICAoYXdhaXQgYSkgKiogYjtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjgsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

```js
/* eslint no-extra-parens: "error"*/

async function foo () {
    (await a) ** b;
}
```

```bash
eslint index.js --fix
```

**What did you expect to happen?**

I believe no error.

By the spec,  [UnaryExpression](https://www.ecma-international.org/ecma-262/10.0/index.html#prod-UnaryExpression) can't be the left-hand side of an [ExponentiationExpression](https://www.ecma-international.org/ecma-262/10.0/index.html#prod-ExponentiationExpression).

`await a ** b` is parsing error for `espree`.

**What actually happened? Please include the actual, raw output from ESLint.**

Auto-fixed to:

```js
/* eslint no-extra-parens: "error"*/

async function foo () {
    await a ** b;
}
```

Message after the fix:

```
4:13  error  Parsing error: Unexpected token **
```

**Are you willing to submit a pull request to fix this bug?**

Yes. 

It would be good to verify is this really a bug in the rule, because `await a ** b` seems to be allowed in Chrome and Firefox. It's interpreted as `(await a) ** b`.

**Comment by kaicataldo** [comment]
I'm also not sure what the correct behavior is here. It looks like Esprima and Babel's parser also allow for `await a ** b`, while TypeScript and Espree throw a parsing error.

**Comment by mdjermanovic** [comment]
Related discussion:

https://github.com/tc39/ecma262/issues/2197

**Comment by mdjermanovic** [comment]
It was decided to leave the spec as-is:

https://github.com/tc39/notes/blob/master/meetings/2020-11/nov-16.md#handling-await-in-left-operands-of-exponentiation

meaning that `await a ** b` is a syntax error, so `no-extra-parens` should not report parens around `await a`.
